### DISCUSSION TITLE
mock service worker enabled in production dotenv inside nextjs template

### DISCUSSION THREAD

**Discussion by Luccasoli** [mswjs/msw#895] [Q&A]
## Describe the bug
[Here](https://github.com/vercel/next.js/blob/canary/examples/with-msw/.env.production) you can see what I described in the title, and in the [documentation](https://mswjs.io/docs/getting-started/integrate/browser#start-worker) it is recommended to keep the service worker disabled to prevent incorrect information (mocked data) for the user.

I'm now starting to study msw, sorry if it's a false bug

## Environment
[Next js template with msw](https://github.com/vercel/next.js/tree/canary/examples/with-msw)

## Expected behavior
the value should be `disabled`

**Comment by Luccasoli** [comment]
If it is indeed an issue, I would love to resolve it! 😊

**Comment by Luccasoli** [comment]
I just noticed in the readme that this was intentional, but keeping it enabled in a production environment can't end up causing some first release issue?

![Captura de tela de 2021-08-29 21-05-21](https://user-images.githubusercontent.com/31039396/131269815-b09914e9-d2c7-4192-8e8d-f93971a7a3fd.png)

**Comment by kettanaito** [comment]
Hey, @Luccasoli. Thank you for raising this, it's a great question.

The documentation is correct: MSW should generally not be enabled in production, as there's very little reason for using mocks in production.

In the case of NextJS, the intention was to keep the environmental variable controlling MSW enabled so that you can witness MSW working in your preview builds. I don't think that keeping it enabled in the production `.env` is a good solution for that, though. 

I've raised the question of whether NextJS has a preview-specific `.env` file that we could utilize.

**Reply by Luccasoli** [reply]
Got it, thanks for the answer!

About Nextjs `.env`, I even did some tests here to confirm and it only uses/prioritizes `.env.production` in preview environment :(

I really think there should be a `.env.preview` just for the preview... but since it doesn't, wouldn't it be better to disable MSW in this environment? Because I imagine that the idea of the preview would be in fact already testing with the environment integrated with the backend

**Reply by kettanaito** [reply]
I've [raised this](https://github.com/vercel/next.js/pull/20335/files#r699319934) in the pull request that introduced the change. Perhaps it was a typo `enabled -> disabled`, but I'd like to learn more before passing my judgment.
